contracts/Governance/Timelock.sol

```diff
@@ -1,123 +1,160 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.13;
+pragma experimental ABIEncoderV2;
 
-contract Timelock {
-    event NewAdmin(address indexed newAdmin);
-    event NewPendingAdmin(address indexed newPendingAdmin);
-    event NewDelay(uint256 indexed newDelay);
-    event CancelTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
-    event ExecuteTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
-    event QueueTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
+import "@openzeppelin/contracts/access/Ownable.sol";
 
-    uint256 public constant GRACE_PERIOD = 14 days;
-    uint256 public constant MINIMUM_DELAY = 2 days;
-    uint256 public constant MAXIMUM_DELAY = 30 days;
+contract Timelock is Ownable {
+    enum TrnasactionStatus {
+        NonExistent,
+        Queued,
+        Approved
+    }
+
+    struct Transaction {
+        address target;
+        uint256 value;
+        string signature;
+        bytes data;
+    }
+
+    uint256 public minimumDelay;
+    uint256 public maximumDelay;
+    uint256 public gracePeriod;
+
+    mapping(bytes32 => TrnasactionStatus) public queuedTransactions;
 
-    address public admin;
-    address public pendingAdmin;
+    address public proposer;
     uint256 public delay;
```

Comment:
NIT
move it to be next to the max and min delays.. Was looking for it and I missed it since it wasn't close to them